### ISSUE TITLE
Making a deep copy of a backend resets the `id`

### ISSUE THREAD

**Issue by milliams** [ganga-devs/ganga#60]
On `develop`:
```python
import time
j = Job()
j.submit()
time.sleep(2) #Wait for job to start
print j.backend.id
f = j.copy()
print j.backend.id
```
will print
```
84286
-1
```
and so for some reason, copying the job has reset the `id` of the job's backend to the default value. I found this behaviour since it happens during a small step of the copy process, when the backend is deep-copied:
```python
import time, copy
j = Job()
j.submit()
time.sleep(2) #Wait for job to start
print j.backend.id
b = copy.deepcopy(j.backend)
print j.backend.id
```
It should be noted that `b.id` and `f.backend.id` are all set to `-1` as well so the information just seems to be being lost entirely.

Looking at the tests now, it seems it's failing on [Savannah8009](http://ganga.web.cern.ch/ganga/release/6.1.14-pre/reports/latest/output/Ganga.test.Bugs.Savannah8009__localxml.out) which was the test I was converting when I came across this.

**Comment by rob-c** [comment]
@milliams Just updated the develop and release braches so I'll recheck now

My initial reaction to this is:
I would naively expect this information to be lost anyway as this is only set once the job is submitted and shouldn't apply to a new job created from copying an old job.
If the test agrees with what I'm saying then we need to determine what the correct behaviour should be.

**Comment by egede** [comment]
@rob-c Look closer at the original bug. It is the existing job that is losing its id - not the newly created one. So this is clearly a bug.

**Comment by rob-c** [comment]
Ah, on reading I realise that the default value is being assigned to the incorrect object in the copy method.

I'll look into this now as this is a nice concrete example of this happening.

I'm starting to lose sight of the wood for the trees at this point.

**Comment by rob-c** [comment]
OK, after some testing I can't reproduce this and the test is running correctly at the moment.

I think that this bug was caused by the use of setstate which I'm hoping to deprecate(or fix) in 6.1.15 as the current workflow of empty/(empty+cache_data) vs fully initialized GangaObjects is difficult to follow.

As this particular bug appears to have been fixed I propose we can close this unless this is reproducible in your setup @milliams ?

**Comment by milliams** [comment]
This bug does indeed seem to be fixed now.
